server/server.js

```diff
@@ -19,41 +23,46 @@ const distPath = path.resolve(__dirname, '..', 'dist');
 const store = makeStore();
 const RoutingContext = createFactory(RoutingContextComponent);
 const Provider = createFactory(ProviderComponent);
+const verify = (token, callback) => {
+  return jwtoken.verify(token, 'secret', {}, callback);
+};
 
 const MongoDB = mongoose.connect(process.env.MONGOLAB_URI || config.developmentDB).connection;
 MongoDB.on('error', (err) => {
   console.log('Failed to connect go Mongo DB');
   console.log(err.message);
 });
-
 MongoDB.once('open', () => {
   console.log('Connected to Mongo DB');
 });
 
-console.log('Server running...');
-
 app.use(express.static(distPath));
+app.use(cookieParser());
 app.use((req, res) => {
   const location = createLocation(req.url);
+  const posToken = req.cookies._posToken;
 
   match({ routes, location }, (err, redirectLocation, renderProps) => {
-    if (err) {
-      console.error(err);
-      return res.status(500).end('Internal server error');
-    }
-
-    if (!renderProps) return res.status(404).end('Not found');
-
-    const InitialComponent = Provider(
-      { store },
-      RoutingContext(renderProps)
-    );
-
-    const initialState = store.getState();
-
-    const componentHTML = renderToString(InitialComponent);
-
-    const HTML = `
+    verify(posToken, (err, token) => {
+      if (err) {
+        console.error(err);
+        return res.redirect(302, '/');
+      }
+
+      if (!renderProps) return res.status(404).end('Not found');
+
+      const user = {
+        name: 'Fu Manchu'
+      };
+      const initialState = _.merge(store.getState(), { user, token: posToken });
+      const rehydratedStore = makeStore(initialState);
```

Comment:
@achiu8 This is the hacky stuff I was doing. Without this, store.getStore() returns the state without the user (because user is not stored on the server) causing InitialComponent to be the login form since it is checking for the presence of this.props.user. This is causing the login form to appear briefly in the beginning after refreshing the page. It's also causing the react checksum invalid error.